asm/module.go

```diff
@@ -98,6 +99,16 @@ func (gen *generator) indexTopLevelEntities(old *ast.Module) error {
 	return nil
 }
 
+// giveUnnamedIdentID give unnamed variable ID to ensure no conflict with others unnamed variables
+func (gen *generator) giveUnnamedIdentID(ident ir.GlobalIdent, id *int64) ir.GlobalIdent {
+	newIdent := ident
```

Comment:
As ident is passed by value (and not pointer), I think we don't need to duplicate `ident` to `newIdent`.

The proposed changed would look something as follows:

```go
 // giveUnnamedIdentID assigns an unused ID to the global identifier if unnamed.
func giveUnnamedIdentID(ident ir.GlobalIdent, id *int64) ir.GlobalIdent {
	if ident.IsUnnamed() {
		// Assign next unused ID to unnamed global identifier.
		ident.SetID(*id)
		*id++
	}
	return ident
}
```